### PR TITLE
feat: Allow to disable ingress creation for raw deployment mode

### PR DESCRIPTION
**Pull request by terrytangyuan**
We'd like to disable this defer the creation to a third-party controller that configures/creates it correctly to meet additional requirements.

### CODE REVIEW

File: charts/kserve-resources/README.md [hunk 1]
```diff
@@ -66,6 +67,8 @@ $ helm install kserve oci://ghcr.io/kserve/charts/kserve --version v0.12.0-rc1
 | kserve.servingruntime.alibi.image | string | `"kserve/alibi-explainer"` |  |
 | kserve.servingruntime.art.defaultVersion | string | `"v0.12.0-rc1"` |  |
 | kserve.servingruntime.art.image | string | `"kserve/art-explainer"` |  |
+| kserve.servingruntime.huggingfaceserver.image | string | `"kserve/huggingfaceserver"` |  |
```
Comment (danielezonca) [inline charts/kserve-resources/README.md]: Probably leftovers of a different PR

Comment (terrytangyuan) [reply charts/kserve-resources/README.md]: Yeah I am fixing that there https://github.com/kserve/kserve/pull/3437. Will rebase once merged.

File: pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go [hunk 2]
```diff
@@ -271,7 +271,7 @@ func (r *RawIngressReconciler) Reconcile(isvc *v1beta1.InferenceService) error {
 	if r.ingressConfig.IngressDomain == constants.ClusterLocalDomain {
 		isInternal = true
 	}
-	if !isInternal {
+	if !isInternal && !r.ingressConfig.DisableIngressCreation {
```
Comment (israel-hdez) [inline pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go:274]: This works nicely on clean clusters. However, I wonder how much it should be supported flipping configurations.

So, if you start with `disableIngressCreation=false` the controller will create ingresses for newly created ISVCs.

If you later flip it to `disableIngressCreation=true`, the controller will no longer create ingresses for new ISVCs. However, the ingresses for older ISVC will persist and they will no longer be updated. My rationale is that the controller should remove all managed ingresses if the setting is flipped, in order to reach the desired state.

Weirdly, if you start with `disableIngressCreation=true` and later flip to `false`, the controller will create/update all missing Ingresses for older ISVCs at startup. So, behavior is not that consistent.

**This is a non-blocking comment**

File: charts/kserve-resources/templates/configmap.yaml [hunk 3]
```diff
@@ -511,6 +515,7 @@ data:
         "domainTemplate": "{{ .Values.kserve.controller.gateway.domainTemplate }}",
         "urlScheme": "{{ .Values.kserve.controller.gateway.urlScheme }}",
         "disableIstioVirtualHost": {{ .Values.kserve.controller.gateway.disableIstioVirtualHost }}
+        "disableIngressCreation": {{ .Values.kserve.controller.gateway.disableIngressCreation }},
```
Comment (israel-hdez) [inline charts/kserve-resources/templates/configmap.yaml]: Can you also update the Kustomize manifests?

**Review by israel-hdez — COMMENTED** [review]
Kustomize manifests needs to be updated.
The other comment is just non-blocking feedback/discussion.

File: charts/kserve-resources/templates/configmap.yaml (hunk 3 above)
Comment (terrytangyuan) [reply charts/kserve-resources/templates/configmap.yaml]: Thanks. Updated

**Review by israel-hdez — APPROVED** [review]
I approve!

/lgtm

**Comment by terrytangyuan** [comment]
Test failure is unrelated.

/assign @yuzisun

**Comment by yuzisun** [comment]
> Test failure is unrelated.
> 
> /assign @yuzisun

Not sure why the helm test keeps failing

**Comment by terrytangyuan** [comment]
@yuzisun helm-test is passing now. There were some syntax issue in YAML.

**Pull request merged**